deeppavlov/models/ner/NER_model.py

```diff
@@ -0,0 +1,307 @@
+                                         filters=128,
+                                         kernel_size=(1, 3),
+                                         strides=(1, 1),
+                                         padding="same",
+                                         name="conv2",
+                                         kernel_initializer=xavier_initializer_conv2d())
+                char_cnn = tf.reduce_max(conv2, axis=2)
+
+                self._input_features.append(char_cnn)
+
+        # ELMo
+        if elmo_dim is not None:
+            with tf.variable_scope("elmo_emb"):
+                padded_x_tokens_ph = tf.placeholder(tf.string, [None, None], name="padded_x_tokens")
+                self._xs_ph_list.append(padded_x_tokens_ph)
+
+                elmo = hub.Module("https://tfhub.dev/google/elmo/2", trainable=True)
+                emb = elmo(inputs={"tokens": padded_x_tokens_ph, "sequence_len": self.real_sent_lengths_ph},
+                           signature="tokens", as_dict=True)["elmo"]
+                elmo_emb = tf.layers.dense(emb, elmo_dim, activation=None)
+                self._input_features.append(elmo_emb)
+
+        features = tf.nn.dropout(tf.concat(self._input_features, axis=2), self._dropout_ph)
+
+        with tf.variable_scope("bi_lstm_words"):
+            cell_fw = tf.contrib.rnn.LSTMCell(lstm_hidden_size)
+            cell_bw = tf.contrib.rnn.LSTMCell(lstm_hidden_size)
+            (output_fw, output_bw), _ = tf.nn.bidirectional_dynamic_rnn(cell_fw, cell_bw, features,
+                                                                        sequence_length=self.real_sent_lengths_ph,
+                                                                        dtype=tf.float32)
+            self.output = tf.concat([output_fw, output_bw], axis=-1)
+
+            ntime_steps = tf.shape(self.output)[1]
+            self.output = tf.reshape(self.output, [-1, 2 * lstm_hidden_size])
+            layer1 = tf.nn.dropout(tf.layers.dense(inputs=self.output, units=lstm_hidden_size, activation=None,
+                                                   kernel_initializer=xavier_initializer()), self._dropout_ph)
+            pred = tf.layers.dense(inputs=layer1, units=n_tags, activation=None,
+                                   kernel_initializer=xavier_initializer())
+            self.logits = tf.reshape(pred, [-1, ntime_steps, n_tags])
+
+            log_likelihood, self.transition_params = tf.contrib.crf.crf_log_likelihood(self.logits,
+                                                                                       self._y_ph,
+                                                                                       self.real_sent_lengths_ph)
+        # loss and opt
+        with tf.variable_scope("loss_and_opt"):
+            self.loss = tf.reduce_mean(-log_likelihood)
+            self.train_op = self.get_train_op(self.loss)
+
+        self.predict = self.predict_crf
```

Comment:
Are there any other cases? Maybe you can just rename `predict_crf` to `predict`? Or use `self.predict_crf` instead of `self.predict` everywhere?

Reply:
At first, I wanted to you CRF or a Bi-LSTM instead. Fixed!